compiler/rustc_trait_selection/src/traits/object_safety.rs

```diff
@@ -348,6 +352,17 @@ fn predicate_references_self<'tcx>(
     }
 }
 
+fn super_predicates_have_non_lifetime_binders(
+    tcx: TyCtxt<'_>,
+    trait_def_id: DefId,
+) -> SmallVec<[Span; 1]> {
```

Comment:
Perf: I guess we could bail early here if `!tcx.features().non_lifetime_binders`...

Reply:
Seems reasonable to me.